index.js

```diff
@@ -581,15 +581,17 @@ var torrentStream = function(link, opts, cb) {
 		if (engine.bitfield) wire.bitfield(engine.bitfield);
 	});
 
-	swarm.pause();
+	var pauseSwarm = true;
 
 	if (link.files && engine.metadata) {
 		swarm.resume();
+		pauseSwarm = false;
 		ontorrent(link);
 	} else {
 		fs.readFile(torrentPath, function(_, buf) {
 			if (destroyed) return;
 			swarm.resume();
+			pauseSwarm = false;
```

Comment:
Agreed, don't know how I missed that.

Do you think that moving ``pauseSwarm = false;`` above ``fs.readFile()`` could create any issues, can ``torrentPath`` get to this point and not be a valid path?